utils/cliutils/utils.go

```diff
@@ -314,3 +316,84 @@ func CreateServerDetailsFromFlags(c *cli.Context) (details *config.ServerDetails
 func IsLegacyGoPublish(c *cli.Context) bool {
 	return c.Command.Name == "go-publish" && c.NArg() > 1
 }
+
+func offerConfig(c *cli.Context) (*coreConfig.ServerDetails, error) {
+	confirmed, err := ShouldOfferConfig()
+	if !confirmed || err != nil {
+		return nil, err
+	}
+	details := createServerDetailsFromFlags(c)
+	configCmd := coreCommonCommands.NewConfigCommand().SetDefaultDetails(details).SetInteractive(true).SetEncPassword(true)
+	err = configCmd.Config()
+	if err != nil {
+		return nil, err
+	}
+
+	return configCmd.ServerDetails()
+}
+
+func CreateServerDetailsWithConfigOffer(c *cli.Context, excludeRefreshableTokens bool) (*coreConfig.ServerDetails, error) {
+	createdDetails, err := offerConfig(c)
+	if err != nil {
+		return nil, err
+	}
+	if createdDetails != nil {
+		return createdDetails, err
+	}
+
+	details := createServerDetailsFromFlags(c)
+	// If urls or credentials were passed as options, use options as they are.
+	// For security reasons, we'd like to avoid using part of the connection details from command options and the rest from the config.
+	// Either use command options only or config only.
+	if credentialsChanged(details) {
+		return details, nil
+	}
+
+	// Else, use details from config for requested serverId, or for default server if empty.
+	confDetails, err := coreCommonCommands.GetConfig(details.ServerId, excludeRefreshableTokens)
+	if err != nil {
+		return nil, err
+	}
+
+	// Take InsecureTls value from options since it is not saved in config.
+	confDetails.InsecureTls = details.InsecureTls
+	confDetails.Url = clientutils.AddTrailingSlashIfNeeded(confDetails.Url)
+	confDetails.DistributionUrl = clientutils.AddTrailingSlashIfNeeded(confDetails.DistributionUrl)
+
+	// Create initial access token if needed.
+	if !excludeRefreshableTokens {
+		err = coreConfig.CreateInitialRefreshableTokensIfNeeded(confDetails)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return confDetails, nil
+}
+
+func createServerDetailsFromFlags(c *cli.Context) (details *coreConfig.ServerDetails) {
+	details = CreateServerDetailsFromFlags(c)
+	details.ArtifactoryUrl = details.Url
+	details.Url = ""
```

Comment:
This logic is only relevant for running command under `rt` scope. 
It allows users to use `--url` for Artifactory commands instead of `--artifactory-url`.
For example:
```
jfrog rt download --url=bla
```
The above command is "translated" to:
```
jfrog rt download --artifactory-url=bla
``` 

In the Xray case, you may allow the following:
```
jfrog xr audit-npm --url=
```
And make sure the URL is treated as an **Xray** URL.